File: Awesome.AI.Source/Awesome.AI/CoreSystems/Monologue1.cs
using Awesome.AI.Core;
using Awesome.AI.Core.Spaces;
using static Awesome.AI.Variables.Enums;

namespace Awesome.AI.CoreSystems
{
    public class Monologue1
    {
        private TheMind mind;
        private Monologue1() { }

        public Monologue1(TheMind mind)
        {
            this.mind = mind;
        }

        public string Result { get; set; }
        public string Subject { get; set; }
        public string Relevance { get; set; }

        private string prev = "im so happy";
        private string curr = "";
        private int counter = 0;

        private string GetRelevance(string str1, string str2)
        {
            string num1 = new string(str1.Where(char.IsDigit).ToArray());
            string num2 = new string(str2.Where(char.IsDigit).ToArray());
            
            try
            {
                if(str1 == "im so happy")
                    return ", but ";

                int _num1 = int.Parse(num1);
                bool upper1 = _num1 > 4;

                int _num2 = int.Parse(num2);
                bool upper2 = _num2 > 4;

                if (upper1 && upper2)
                    return ", and ";

                if (!upper1 && !upper2)
                    return ", and ";

                return ", but ";
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        public void Create(bool _pro)
        {
            MINDS mt;
            string idx = "";
            string sub = "";

            try
            {
                if (!_pro)
                    return;

                counter++;

                if (counter < 2)
                    return;

                counter = 0;

                if (mind.STATE == STATE.QUICKDECISION)
                    return;

                if (mind.unit_actual.IsQUICKDECISION())
                    return;

                if (mind.unit_actual.IsDECISION())
                    return;

                //if (!CONST.SAMPLE20.RandomSample(mind))
                //    return;

                sub = mind.hub.GetSubject(mind.unit_actual) ?? "";
                curr = mind?.unit_actual?.Data ?? "";

                if (sub == "")
                    return;

                Relevance = GetRelevance(prev, curr);

                string res = prev + "||" + curr;
                prev = curr;

                Result = res;
                Subject = sub;
            }
            catch (Exception ex)
            {
                throw;
            }
        }
    }
}

File: Awesome.AI.Python/CoreSystems/Monologue1.py
"""Deterministic inner monologue ported from ``Monologue1.cs``."""
from awesome_ai.Variables.Enums import STATE


class Monologue1:
    def __init__(self, mind):
        self.mind = mind
        self.Result = ""
        self.Subject = ""
        self.Relevance = ""
        self.prev = "im so happy"
        self.curr = ""
        self.counter = 0

    def GetRelevance(self, str1, str2):
        number1 = "".join(character for character in str1 if character.isdigit())
        number2 = "".join(character for character in str2 if character.isdigit())

        if str1 == "im so happy":
            return ", but "

        upper1 = int(number1) > 4
        upper2 = int(number2) > 4
        if upper1 == upper2:
            return ", and "
        return ", but "

    def Create(self, _pro):
        if not _pro:
            return

        self.counter += 1
        if self.counter < 2:
            return
        self.counter = 0

        if self.mind.STATE == STATE.QUICKDECISION:
            return

        unit = self.mind.unit_actual
        if unit.IsQDECISION() or unit.IsDECISION():
            return

        subject = self.mind.hub.GetSubject(unit) or ""
        self.curr = unit.Data if unit is not None else ""

        if subject in ("", "init"):
            return

        self.Relevance = self.GetRelevance(self.prev, self.curr)
        result = self.prev + "||" + self.curr
        self.prev = self.curr
        self.Result = result
        self.Subject = subject
